canvas_cli/tests.py

```diff
@@ -1,11 +1,170 @@
+    return [
+        ("List empty plugins", "list", 0, [], [f"{plugin_name}"]),
+        (
+            "Install new plugin",
+            f"install {plugin_name}",
+            0,
+            [
+                f"Plugin {plugin_name} has a valid CANVAS_MANIFEST.json file",
+                "Installing plugin:",
+                "Posting",
+                f"Plugin {plugin_name} successfully installed!",
+            ],
+            [],
+        ),
+        ("List newly installed plugin", "list", 0, [f"{plugin_name}@0.0.1	enabled"], []),
+        (
+            "Disable plugin",
+            f"disable {plugin_name}",
+            0,
+            [f"Disabling {plugin_name} using ", f"Plugin {plugin_name} successfully disabled!"],
+            [],
+        ),
+        ("List disabled plugin", "list", 0, [f"{plugin_name}@0.0.1	disabled"], []),
+        (
+            "Enable plugin",
+            f"enable {plugin_name}",
+            0,
+            [f"Enabling {plugin_name} using ", f"Plugin {plugin_name} successfully enabled!"],
+            [],
+        ),
+        (
+            "Uninstall enabled plugin",
+            f"uninstall {plugin_name}",
+            1,
+            [
+                f"Uninstalling {plugin_name} using",
+                'Status code 403: {"detail":"Cannot delete an enabled plugin."}',
+            ],
+            [],
+        ),
+        ("Disable plugin", f"disable {plugin_name}", 0, [], []),
+        (
+            "Uninstall disabled plugin",
+            f"uninstall {plugin_name}",
+            0,
+            [
+                f"Uninstalling {plugin_name} using",
+                f"Plugin {plugin_name} successfully uninstalled!",
+            ],
+            [],
+        ),
+        ("List empty plugins again", "list", 0, [], [f"{plugin_name}"]),
+    ]
+
+
+@patch("keyring.get_password")
+@patch("keyring.set_password")
+@pytest.mark.integtest
+def test_canvas_list_install_disable_enable_uninstall(
+    mock_get_password: MagicMock,
+    mock_set_password: MagicMock,
+    plugin_name: str,
+    cli_test_steps: list[tuple[str, str, int, list[str], list[str]]],
+) -> None:
+    """Tests that the Canvas CLI can list, install, disable, enable, and uninstall a plugin"""
+
+    mock_get_password.return_value = None
+    mock_set_password.return_value = None
+
+    runner.invoke(app, "init", input=plugin_name)
+    protocol = open(f"./{plugin_name}/protocols/my_protocol.py", "w")
+    p = [
+        "from canvas_sdk.events import EventType\n",
+        "from canvas_sdk.protocols import BaseProtocol\n",
+        "from logger import log\n",
+        "\n",
+        "\n",
+        "class Protocol(BaseProtocol):\n",
+        "\tRESPONDS_TO = EventType.Name(EventType.ASSESS_COMMAND__CONDITION_SELECTED)\n",
+        """\tNARRATIVE_STRING = "I was inserted from my plugin's protocol."\n""",
+        "\n",
+        "\tdef compute(self):\n",
+        "\t\tlog.info(self.NARRATIVE_STRING)\n",
+        "\t\treturn []\n",
+    ]
```

Comment:
This might be more readable with multiline triple quotes. You could then use `write` instead of `writelines`.